ProjectVoyage/View Controllers/Chat message/MessagesVC.m

```diff
@@ -0,0 +1,340 @@
+    return 40.0;
+}
+
+- (NSString *)tableView:(UITableView *)tableView titleForHeaderInSection:(NSInteger)section {
+    return @"test";
+}
+
+- (UIView *)tableView:(UITableView *)tableView viewForHeaderInSection:(NSInteger)section {
+    CGRect frame = CGRectMake(0, 0, tableView.frame.size.width, 40);
+    
+    UIView *view = [[UIView alloc] initWithFrame:frame];
+    view.backgroundColor = [UIColor clearColor];
+    view.autoresizingMask = UIViewAutoresizingFlexibleWidth;
+    
+    UILabel *label = [[UILabel alloc] init];
+    label.text = [self tableView:tableView titleForHeaderInSection:section];
+    label.textAlignment = NSTextAlignmentCenter;
+    label.font = [UIFont fontWithName:@"Helvetica" size:20.0];
+    [label sizeToFit];
+    label.center = view.center;
+    label.font = [UIFont fontWithName:@"Helvetica" size:13.0];
+    label.backgroundColor = [UIColor colorWithRed:207/255.0 green:220/255.0 blue:252.0/255.0 alpha:1];
+    label.layer.cornerRadius = 10;
+    label.layer.masksToBounds = YES;
+    label.autoresizingMask = UIViewAutoresizingNone;
+    [view addSubview:label];
+    
+    return view;
+}
+
+
+- (void)tableViewScrollToBottomAnimated:(BOOL)animated {
+    NSInteger numberOfRows = self.chat.messages.count;
+    
+    // Open chat in the very last message
+    if (numberOfRows) {
+        NSIndexPath *indexPath = [NSIndexPath indexPathForRow:(self.chat.messages.count - 1) inSection:0];
+        
+        [_tableView scrollToRowAtIndexPath:indexPath
+                    atScrollPosition:UITableViewScrollPositionBottom
+                    animated:animated];
+    }
+}
+
+
+#pragma mark - InputbarDelegate
+
+-(void)inputbarDidPressRightButton:(Inputbar *)inputbar {
+    Message *message = [[Message alloc] init];
+    message.text = [Util removeEndSpaceFrom:inputbar.text];
+    
+    //Store Message in memory
+    [self.chat.messages addObject:message];
+    
+    //Insert Message in UI
+    NSIndexPath *indexPath = [NSIndexPath indexPathForRow:self.chat.messages.count - 1 inSection:0];
+    
+    [self.tableView beginUpdates];
+    
+    [self.tableView insertRowsAtIndexPaths:@[indexPath] withRowAnimation:UITableViewRowAnimationBottom];
+    
+    [self.tableView endUpdates];
+    
+    [self.tableView scrollToRowAtIndexPath:[NSIndexPath indexPathForRow:self.chat.messages.count-1 inSection:0]
+                    atScrollPosition:UITableViewScrollPositionBottom
+                    animated:YES];
+    
+    //Send message to server
+    //[self.gateway sendMessage:message];
+    
+    /*
+    PFQuery *query = [PFQuery queryWithClassName:@"Chat"];
+    [query getObjectInBackgroundWithId:_chat.objectId
+                                 block:^(PFObject *chat, NSError *error) {
+        if (error) {
+            NSLog(@"%@", error);
+        }
+        NSLog(@"%@", chat);
+        chat[@"messages"] = (NSArray *)self.chat.messages;
+        [chat saveInBackground];
+    }];
+     */
```

Comment:
delete unused code

Reply:
Thanks for the feedback, I'm curious to gain some advice here on the advantages of deleting unused code in my commits.

Those pieces of code are usually drafts of the stuff I was trying to implement but couldn't at the moment because I was implementing previous commits that would eventually lead to a working implementation of that.

In that case, is it better practice to put those snippets in an uncommitted text file as a reference?